app/lib/actions/katello/pulp_selector.rb

```diff
@@ -1,12 +1,30 @@
 module Actions
   module Katello
     class PulpSelector < Actions::Base
+      middleware.use Actions::Middleware::PropagateOutput
       def plan(backend_actions, repository, smart_proxy, *args)
+        found_action = PulpSelector.select(backend_actions, repository, smart_proxy)
+        fail "Could not locate an action for type #{backend_type}" unless found_action
+        sequence do
+          action= plan_action(found_action, repository, smart_proxy, *args)
+          begin
+            action_output = action.output
+          rescue
+            plan_self
+          else
+            plan_self(:subaction_output => action_output)
+          end
+        end
```

Comment:
Not sure if this is the best way to handle this, but unless there is an output to propagate, output is not available therefore `action.output` throws an error saying plan_self has to called before accessing output which is handled by the rescue block here which plans self without any sub_action_output. In actions where propagation is required we do have a plan_self with the sub_action_output.

Reply:
I played around with this and i see what you mean, we need a run method for this to work.  A couple of options i can think of:

1)  Create a module, something called 'ActionWithPropagatedOutput', which looks like this:
```
module ActionWithPropagatedOutput
   included do 
           middleware.use Actions::Middleware::PropagateOutput
   end

  def run
    #empty run method
  end
end
```

Then instead of specifying the middlware, we include this module.  

2) do similar to above, but use a class we inherit from. 

I think i like the module route better for this.

Reply:
@jlsherrill : The module way looks better and works well. Pushed a new commit with the changes.